src/jit/morph.cpp

```diff
@@ -16649,73 +16638,64 @@ void Compiler::fgMorph()
 
     objectAllocator.Run();
 
-    /* Add any internal blocks/trees we may need */
-
-    fgAddInternal();
-
-#ifdef DEBUG
-    /* Inliner could add basic blocks. Check that the flowgraph data is up-to-date */
-    fgDebugCheckBBlist(false, false);
-    /* Inliner could clone some trees. */
-    fgDebugCheckNodesUniqueness();
-#endif // DEBUG
-
-    fgRemoveEmptyTry();
-
-    EndPhase(PHASE_EMPTY_TRY);
-
-    fgRemoveEmptyFinally();
-
-    EndPhase(PHASE_EMPTY_FINALLY);
+    // Add any internal blocks/trees we may need
+    //
+    DoPhase(this, PHASE_MORPH_ADDFG, &Compiler::fgAddInternal, PhaseChecks::NONE);
```

Comment:
Did you remove the checking here (and in a few other places) on purpose or is this work in progress?

Reply:
I didn't intentionally remove any checks. I may ah. The call to  `Compiler::EndPhase` doesn't trigger checking, just timing and some specialized dumping. So I think the change above is check-neutral -- no checks happen either way.

But I also haven't looked at checking systematically yet. There is probably some checking we could/should do early on, before we have pred lists.